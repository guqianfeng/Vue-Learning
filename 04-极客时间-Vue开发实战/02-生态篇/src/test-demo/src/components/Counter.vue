<template>
  <div>
    <span>{{ count }}</span>
    <button @click="clickHandle">count++</button>
  </div>
</template>

<script>
export default {
  data () {
    return {
      count: 1
    }
  },
  methods: {
    clickHandle () {
      this.count++;
      this.$emit('click', this.count)
    }
  }
}
</script>

<style>

</style>