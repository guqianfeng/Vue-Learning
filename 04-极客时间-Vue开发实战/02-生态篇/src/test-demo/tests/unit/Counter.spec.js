import { mount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'
import sinon from 'sinon'

describe('Counter.vue', () => {
  const click = sinon.spy()
  const wrapper = mount(Counter, {
    listeners: {
      click
    }
  })
  it('renders counter html', () => {
    expect(wrapper.html()).toMatchSnapshot()
  });
  it('count++', () => {
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.count).toBe(2)
    expect(click.called).toBe(true)
    button.trigger('click')
    expect(click.callCount).toBe(2)
  })
})
