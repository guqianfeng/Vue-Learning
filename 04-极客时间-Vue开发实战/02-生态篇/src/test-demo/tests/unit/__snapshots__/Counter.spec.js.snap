// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders counter html 1`] = `<div><span>1</span> <button>count++</button></div>`;

exports[`Counter.vue renders props.msg when passed 1`] = `<div><span>1</span> <button>count++</button></div>`;
